cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -9,10 +9,10 @@ private import semmle.code.cpp.models.interfaces.DataFlow
 cached
 private newtype TNode =
   TExprNode(Expr e) or
-  TParameterNode(Parameter p) { exists(p.getFunction().getBlock()) } or
-  TDefinitionByReferenceNode(VariableAccess va, Expr argument) {
-    definitionByReference(va, argument)
-  } or
+  TPartialDefinitionNode(PartialDefinition pd) or
+  TPostConstructorCallNode(ConstructorCall call) or
```

Comment:
The implicit node associated with a constructor should be pre-update with its post-update node being the corresponding TExprNode.